### PR TITLE
Added a default boundary map and configuration options to WorldAtlasGenerator

### PR DESCRIPTION
**Pull request by jwpgage**
### Description:

Added a default boundary map to `WorldAtlasGenerator`, and changed the required flag to optional for the country boundary map parameter. This allows `WorldAtlasGenerator` to be used to create boundary maps from pbf.

This PR also adds the ability to set custom node, relation, and way filters in `WorldAtlasGenerator`, which was also necessary for border file generation.

### Potential Impact:

The default country boundary map provided is set to the width of the world, and a fake country "AAA". Therefore a country boundary map parameter should still be passed in most use cases.

### Unit Test Approach:

N/A

### Test Results:

Tested default boundary map as a part of country boundary map generation.

------

In doubt: [Contributing Guidelines](https://github.com/osmlab/atlas/blob/dev/CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/generator/world/WorldAtlasGenerator.java [hunk 1]
```diff
@@ -59,6 +64,20 @@
...
+            "osmPbfNodeConfiguration",
+            "The path to the configuration file that defines which PBF Nodes becomes an Atlas Entity.",
+            File::new, Optionality.OPTIONAL);
+    public static final Switch<File> PBF_RELATION_CONFIGURATION = new Switch<>(
+            "osmPbfRelationConfiguration",
+            "The path to the configuration file that defines which PBF Relations becomes an Atlas Entity",
+            File::new, Optionality.OPTIONAL);
+    // filter that does no filtering
+    private static final ConfiguredTaggableFilter PBF_NO_FILTER_CONFIGURATION = new ConfiguredTaggableFilter(
+            new StandardConfiguration(new StringResource("{\n    \"filters\": [\n\n    ]\n}")));
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/generator/world/WorldAtlasGenerator.java]: Here the `\n`s are not necessary! `"{\"filters\": []}"` should work I think.

Comment (lucaspcram) [reply src/main/java/org/openstreetmap/atlas/generator/world/WorldAtlasGenerator.java]: I'm out of the loop here. Is this JSON?

File: src/main/java/org/openstreetmap/atlas/generator/world/WorldAtlasGenerator.java [hunk 2]
```diff
@@ -68,7 +87,16 @@ public static void main(final String[] args)
     @Override
     protected int onRun(final CommandMap command)
     {
-        final CountryBoundaryMap countryBoundaryMap = (CountryBoundaryMap) command.get(BOUNDARIES);
+        CountryBoundaryMap countryBoundaryMap = (CountryBoundaryMap) command.get(BOUNDARIES);
+        if (countryBoundaryMap == null)
+        {
+            final StringResource wholeWorld = new StringResource(
+                    "AAA||" + Rectangle.MAXIMUM.toWkt());
+            countryBoundaryMap = CountryBoundaryMap.fromPlainText(wholeWorld);
```
Comment (lucaspcram) [inline src/main/java/org/openstreetmap/atlas/generator/world/WorldAtlasGenerator.java]: Clever!

File: src/integrationTest/java/org/openstreetmap/atlas/generator/world/WorldAtlasGeneratorIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.apache.hadoop.fs.Path;
+import org.junit.Assert;
+import org.junit.Test;
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.generator.AtlasGeneratorIntegrationTest;
+import org.openstreetmap.atlas.generator.tools.streaming.ResourceFileSystem;
+import org.openstreetmap.atlas.streaming.resource.InputStreamResource;
+
+public class WorldAtlasGeneratorIntegrationTest
+{
```
Comment (lucaspcram) [inline src/integrationTest/java/org/openstreetmap/atlas/generator/world/WorldAtlasGeneratorIntegrationTest.java]: Nice integration test!

**Pull request merged**